packages/sil-voting-app/src/store/proposals/actions.ts

```diff
@@ -1,15 +1,23 @@
 import { Proposal, ProposalStatus, VoteOption } from "@iov/bns";
+import { Governor } from "@iov/bns-governance";
 
-import { getDummyElectionRules, getDummyProposals, getDummyVote } from "./dummyData";
+import { getDummyVote } from "./dummyData";
 import { AddProposalsActionType, ProposalsState } from "./reducer";
 
+let governor: Governor | undefined;
```

Comment:
This module level storage is a suplicate of redux's job. Here you find a proposed alternative using redux: https://github.com/iov-one/ponferrada/pull/520

Reply:
The proposed alternative of #520 always results in `governor === undefined`. I have modified it a bit, storing a `governor` variable local to the `Login` component and setting it in `isExtensionConnected()`.

Reply:
I just looked into it again. What you say makes sense. In https://github.com/iov-one/ponferrada/pull/520/files, the `gonernor` variable is set once in creation of the Login component from the current redux store value (which is undefined at Login creation time). We have to requery the store after `isExtensionConnected` is called. Anyways, your solution is a good step in the right direction and we can always improve things.